### PR TITLE
ISPN-10040 Embedded and server thread pool defaults should be the same

### PR DESCRIPTION
**Pull request by danberindei**
https://issues.jboss.org/browse/ISPN-10040

* Server thread pools now read from embedded

### CODE REVIEW

**Comment by tristantarrant** [comment]
[2019-03-29T08:05:12.636Z] [ERROR] /home/infinispan/workspace/Infinispan_PR-6816/server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/ThreadPoolResource.java:39:8: Unused import - org.infinispan.factories.KnownComponentNames. [UnusedImports]

**Comment by danberindei** [comment]
Updated. Sorry, I thought IntelliJ would remove the import when I added the static import for the new method :)

File: server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/ThreadPoolResource.java [hunk 1]
```diff
@@ -64,11 +73,17 @@
...
 public enum ThreadPoolResource implements ResourceDefinition, ThreadPoolDefinition {
 
-    ASYNC_OPERATIONS("async-operations", 25, 25, 1000, 60000),
-    LISTENER("listener", 1, 1, 100000, 60000),
-    REMOTE_COMMAND("remote-command", 25, 25, 100000, 60000),
-    STATE_TRANSFER("state-transfer", 1, 60, 0, 60000),
-    TRANSPORT("transport", 25, 25, 100000, 60000),
+   ASYNC_OPERATIONS("async-operations", getDefaultMinThreads(ASYNC_OPERATIONS_EXECUTOR),
+                    getDefaultThreads(ASYNC_OPERATIONS_EXECUTOR), getDefaultQueueSize(ASYNC_OPERATIONS_EXECUTOR),
+                    60000),
```
Comment (karesti) [inline server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/ThreadPoolResource.java]: What about adding a constant for 60000 ?

Comment (karesti) [reply server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/ThreadPoolResource.java]: @danberindei ?

Comment (danberindei) [reply server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/ThreadPoolResource.java]: Ok, I can do that

**Comment by danberindei** [comment]
@karesti updated to introduce a constant for the keepalive timeout in `BlockingThreadPoolExecutorFactory`

**Pull request merged**